pkg/scalers/aws_sqs_queue_scaler.go

```diff
@@ -52,25 +61,47 @@ func parseAwsSqsQueueMetadata(metadata, resolvedEnv map[string]string) (*awsSqsQ
 		}
 	}
 
-	if val, ok := metadata["queueURL"]; ok {
-		queueURL := val
+	if val, ok := metadata["queueURL"]; ok && val != "" {
+		meta.queueURL = val
+	} else {
+		return nil, fmt.Errorf("no queueURL given")
+	}
 
-		if len(val) <= 0 {
-			log.Errorf("Error parsing SQS queue metadata %s: %s", "queueURL", errors.New("Empty queueURL is not valid"))
-		} else {
-			meta.queueURL = queueURL
-		}
+	if val, ok := metadata["region"]; ok && val != "" {
+		meta.region = val
+	} else {
+		return nil, fmt.Errorf("no region given")
+	}
+
+	if val, ok := resolvedEnv["awsAccessKeyID"]; ok && val != "" {
+		meta.awsAccessKeyID = val
+	} else {
+		meta.awsAccessKeyID = awsAccessKeyIDEnvVar
+	}
+    var keyName string
+	if keyName, ok := metadata["awsAccessKeyID"]; !ok || keyName == "" {
+		keyName = awsAccessKeyIDEnvVar
+	}
+	
+	if val, ok := resolvedEnv[keyName]; ok && val != "" {
+		meta.awsAccessKeyID = val
+	} else {
+		return nil, fmt.Errorf("cannot find awsAccessKeyId named %s in pod environment", keyName)
+	}
+	if val, ok := resolvedEnv["awsSecretAccessKey"]; ok && val != "" {
+		meta.awsSecretAccessKey = val
+	} else {
+		meta.awsSecretAccessKey = awsSecretAccessKeyEnvVar
```

Comment:
same as with the `awsAccessKeyId` above, get the value from resolvedEnv based on the env var name.

Reply:
done